File: 7_reverse_integer/reverse.py
class Solution:
    def reverse(self, x: int) -> int:
        xs = str(x)
        xs = xs[::-1]

        if xs[-1] == '-':
            xs = '-' + xs[:-1]

        res = int(xs)

        if res > pow(2, 31) - 1 or res < -pow(2, 31) - 1:
            return 0
        return res

File: 7_reverse_integer/reverse.go
package reverseinteger

func reverse(x int) int {
	sign := 1

	if x < 0 {
		x = -x
		sign = -1
	}

	res := 0

	for x > 0 {
		digit := x % 10
		x = x / 10

		if res > 2147483647/10 {
			return 0
		}

		res = res*10 + digit
	}

	return res * sign
}